crates/re_ui/src/list_item.rs

```diff
@@ -315,11 +315,11 @@ impl<'a> ListItem<'a> {
     pub fn show_hierarchical_with_content<R>(
         mut self,
         ui: &mut Ui,
-        id: impl Into<egui::Id>,
+        id_source: impl std::hash::Hash,
         default_open: bool,
         add_body: impl FnOnce(&ReUi, &mut egui::Ui) -> R,
     ) -> ShowCollapsingResponse<R> {
-        let id = id.into();
+        let id = ui.id().with(id_source);
```

Comment:
I think we can't do that. It will break the entire `Expend/Collapse All` because we're depending on the collapsedness to be stored against a "known" id (matched to the `CollapsedScope` stuff).

Reply:
Confirmed: Expend all is broken in the web build.

Reply:
(Well, this is legacy list item, with is gone on main. But the same applies for list item2)

Reply:
The old code created ID clashes when multiple things were selected in the selection panel, since they used the same ID.

I guess we should be explicit and hand an `Id` here, and then for the blueprint view we use a globally unique id (old code), and almost anywhere else the user can pass in `ui.id().with(…)`.

A bit verbose maybe, but that's the only solution I can see.

Reply:
Ok, fixed!